public/projects/umbrella_today.md

```diff
@@ -0,0 +1,43 @@
+# Umbrella Today
+
+Once upon a time, there was a real live app called [Umbrella Today](https://robots.thoughtbot.com/umbrella-today). It was the best weather forecasting app ever invented. It worked like this:
+
+Every morning at 7:00 am, if it was going to rain that day, it sent you a text message saying "Umbrella today!"
+
+![umbrella today 1](umbrella-today-1.png)
+
+Sadly, the app has now gone the way of all bits. But we can rebuild it!
+
+## Discussion
+
+We will build this app in several phases:
+
+**Command-line only** The user enters their zip code; the app calls a weather forecast API and parses the result; the app then returns one of three answers:
+
+1. _yes_ if there is rain in the forecast
+2. _no_ if there isn't
+3. _error_ if there was a problem connecting to the weather API (after a reasonable timeout) (and making sure to *print* a human-readable error message from the response if possible)
+
+Once you figure out how to call the API from JavaScript, the most difficult part will likely be dealing with failure cases.
+
+OpenWeatherMap and WUnderground are two weather forecasting services you could use. Research them and choose one.
+
+* http://openweathermap.org/api
```

Comment:
Darksky.net may also be a viable option for an API

Reply:
👍